packages/perseus/src/widgets/interactive-graphs/mafs-graph.test.tsx

```diff
@@ -63,3 +68,192 @@ describe("MafsGraph", () => {
         expect(mockChangeHandler).toHaveBeenCalled();
     });
 });
+
+function map(
+    num: number,
+    startMin: number,
+    startMax: number,
+    targetMin: number,
+    targetMax: number,
+): number {
+    return (
+        ((num - startMin) / (startMax - startMin)) * (targetMax - targetMin) +
+        targetMin
+    );
+}
+
+describe("MafsGraph", () => {
+    let userEvent: UserEvent;
+    beforeEach(() => {
+        userEvent = userEventLib.setup({
+            advanceTimers: jest.advanceTimersByTime,
+        });
+    });
+
+    it("renders", () => {
+        const mockDispatch = jest.fn();
+        const state: InteractiveGraphState = {
+            type: "segment",
+            hasBeenInteractedWith: true,
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+            snapStep: [0.5, 0.5],
+            coords: [
+                [
+                    [0, 0],
+                    [-7, 0.5],
+                ],
+            ],
+        };
+
+        const baseMafsGraphProps = getBaseMafsGraphProps();
+
+        render(
+            <MafsGraph
+                state={state}
+                dispatch={mockDispatch}
+                {...baseMafsGraphProps}
+            />,
+        );
+
+        const line = screen.getByTestId("movable-line__line");
+
+        expect(line).toBeInTheDocument();
+        expect(line.getAttribute("x1")).toBe("0");
+        expect(line.getAttribute("y1")).toBe("0");
+
+        // Map graph coordinates to SVG coordinates
+        const halfXBox = baseMafsGraphProps.box[0] / 2;
+        const halfYBox = baseMafsGraphProps.box[1] / 2;
+        const expectedX2 = map(-7, -10, 10, halfXBox * -1, halfXBox);
+        const expectedY2 = map(0.5, -10, 10, halfYBox * -1, halfYBox) * -1;
```

Comment:
You'll see this a lot in the tests. Basically I'm converting graph space (-10 to 10) to pixel space (-200 to 200 when width/height is 400).

I think it's equally valuable to be testing logic numbers (graph space, the product of the reducer) and real numbers (pixel space, the product of actual SVG rendering).

Reply:
I answered this in another comment, but I ended up using helpers from Mafs for this.